numpy/lib/function_base.py

```diff
@@ -3948,7 +3948,7 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
             r_above = r_above.squeeze(0)
             r_below = r_below.squeeze(0)
 
-        r_above[r_above < 0.5] = r_below
+        r_above[r_above < 0.5] = r_below[r_above < 0.5]
```

Comment:
Opst, good point the `np.copyto(r_above, r_below, where=r_above < 0.5)` call is better in this case then I think.

Reply:
No, it actually caught a warning because I wrongly used `r_above` instead of `weights_above` ant it was containing some `np.nan`